### ISSUE TITLE
Town-River-Swamp Mess

### ISSUE THREAD

**Issue by CoroNaut** [CleverRaven/Cataclysm-DDA#21241]
[Build 6504] I honestly don't know if this is a bug or if it is just the mapgen taking the day off. I have no educated guess to give as to what went on here:

Map picture:

![capture](https://user-images.githubusercontent.com/24307650/27144972-d9c8442e-5101-11e7-8d70-174cb21c24a9.PNG)

Ground level picture:

![capture](https://user-images.githubusercontent.com/24307650/27145015-fe0ec9ac-5101-11e7-8486-dfac297c54c6.PNG)

**Comment by CoroNaut** [comment]
Another weird but interesting mapgen problem:
Ground view:

![capture](https://user-images.githubusercontent.com/24307650/27190993-66ff9f0a-51c4-11e7-9973-5bd25f6d47ba.PNG)

Map view (center of screen is gas station in the middle of the road... while in the middle of the water):
![capture](https://user-images.githubusercontent.com/24307650/27191031-87d48a2e-51c4-11e7-8659-0606e276094f.PNG)

**Comment by Treah** [comment]
LOL wow thats some interesting mapgen right there. My guess with the gasstation is that it wanted to spawn it on the side but there was no ground there so it spawned it in the middle of the road. Very edge case and this one is also a Jsonfied version of the gas station. It also happens to be one that I modified for spacing. I however kept the same size as the original so I don't think any of the changes I made would have caused this.
